### ISSUE TITLE
Json generated by sprinfox contains equivalent endpoints for different operations that gives error while parsing with swagger-ui

### ISSUE THREAD

**Issue by forever-programmer** [springfox/springfox#1790]
In my spring-mvc project I've two equivalent endpoints with different operations.

Say 
http://api.myhost.com/users/{userId} -- GET to fetch user
http://api.myhost.com/users/{userId} -- PUT to update user

Now the json generated by springfox has two separate entries for these endpoints. One for get and one for put.

`"paths": {
			"/users/{userId}": {
				"get": {
					"tags": ["my-controller"],
					"summary": "getUser",
					"operationId": "getUserUsingGET",
					"consumes": ["application/json"],
					"produces": ["*/*"],
					"parameters": [{
						"name": "userId",
						"in": "path",
						"description": "userId",
						"required": true,
						"type": "string"
					}],
					"responses": {
						"200": {
							"description": "OK",
							"schema": {
								"$ref": "#/definitions/ApiResponseDtoÂ«booleanÂ»"
							}
						},
						"401": {
							"description": "Unauthorized"
						},
						"403": {
							"description": "Forbidden"
						},
						"404": {
							"description": "Not Found"
						}
					}
				}
			},
			"/users/{userId}": {
				"put": {
					"tags": ["my-controller"],
					"summary": "updateUser",
					"operationId": "updateUserUsingPOST_1",
					"consumes": ["application/json"],
					"produces": ["*/*"],
					"parameters": [{
						"name": "userId",
						"in": "path",
						"description": "userId",
						"required": true,
						"type": "string"
					}],
					"responses": {
						"200": {
							"description": "OK",
							"schema": {
								"$ref": "#/definitions/ApiResponseDtoÂ«booleanÂ»"
							}
						},
						"201": {
							"description": "Created"
						},
						"401": {
							"description": "Unauthorized"
						},
						"403": {
							"description": "Forbidden"
						},
						"404": {
							"description": "Not Found"
						}
					}
				},`

This json gets converted to following YAML code.
`paths:
  '/users/{userId}':
    get:
      tags:
        - my-controller
      summary: getUser
      operationId: getUserUsingGET
      consumes:
        - application/json
      produces:
        - '*/*'
      parameters:
        - name: userId
          in: path
          description: userId
          required: true
          type: string
      responses:
        '200':
          description: OK
          schema:
            $ref: '#/definitions/ApiResponseDtoÂ«booleanÂ»'
        '401':
          description: Unauthorized
        '403':
          description: Forbidden
        '404':
          description: Not Found
  '/users/{userId}':
    put:
      tags:
        - my-controller
      summary: updateUser
      operationId: updateUserUsingPOST_1
      consumes:
        - application/json
      produces:
        - '*/*'
      parameters:
        - name: userId
          in: path
          description: userId
          required: true
          type: string
      responses:
        '200':
          description: OK
          schema:
            $ref: '#/definitions/ApiResponseDtoÂ«booleanÂ»'
        '201':
          description: Created
        '401':
          description: Unauthorized
        '403':
          description: Forbidden
        '404':
          description: Not Found`

When I try to parse this payload with swagger-ui, I get the following error

`Semantic error at paths./users/{userId}
Equivalent paths are not allowed.
`

**Comment by dilipkrish** [comment]
Don't really know how you got two endpoints. Thats not possible via springfox. You should see something like this

```json
'/users/{userId}': {
     "get": {...},
     "put": {...}
 }
```
and not 

```json
'/users/{userId}': {
     "get": {...},
},
'/users/{userId}': {
     "put": {...}
 }
```

**Comment by forever-programmer** [comment]
Thanks for the reply. Let me give you some more details.

In my java code I've 2 methods in my controller, say, `getUserDetails() and updateUserDetails()`.

They are as follows

`@RequestMapping(value = "/users/{userId}", method = {RequestMethod.PUT})
@ResponseStatus(HttpStatus.OK)
public ApiResponseDto<boolean> updateUserDetails(@PathVariable String userId, @RequestBody UserDto userDto)

@RequestMapping(value = "users/{userId}", method = RequestMethod.GET)
@ResponseStatus(HttpStatus.OK)
public ApiResponseDto<List<boolean>> getUserDetails(@PathVariable String userId) `

**Comment by forever-programmer** [comment]
any clue for this issue?

**Comment by dilipkrish** [comment]
not had a chance to look at this

**Comment by forever-programmer** [comment]
Hi.
Can I get some more help here, please.

Thanks.

**Comment by dilipkrish** [comment]
Haven't had a look at this issue

**Comment by dilipkrish** [comment]
This should be fixed in 2.7.0
